src/selectors/profile.js

```diff
@@ -139,6 +139,64 @@ export const getCounterSelectors = (index: CounterIndex): CounterSelectors => {
   return selectors;
 };
 
+const getThreadsForPIDs = createSelector(
+  getThreads,
+  (threads) => {
+    let pid2threads = {};
+    for (let thread of threads) {
+      if (!(thread.pid in pid2threads)) {
+        pid2threads[thread.pid] = [];
+      }
+      pid2threads[thread.pid].push(thread);
+    }
+    return pid2threads;
+  }
+);
+
+export const getIPCCorrelations = createSelector(
+  getThreads,
+  getThreadsForPIDs,
+  (threads, pid2threads) => {
+    function findRecipientMarker(srcData) {
+      if (!(srcData.otherpid in pid2threads)) {
+        throw new Error('recipient not found');
+      }
+      for (let thread of pid2threads[srcData.otherpid]) {
+        for (let i = 0; i < thread.markers.length; i++) {
```

Comment:
We deal with lots of indexes in the profiler, so I usually find it's helpful to name the indexes as I use them, especially as this one is getting passed into another data structure. I would rename `i` to `markerIndex`.